src/PhpDoc/TypeNodeResolver.php

```diff
@@ -263,6 +305,25 @@ private function resolveIdentifierTypeNode(IdentifierTypeNode $typeNode, NameSco
 		return new ObjectType($stringName);
 	}
 
+	private function tryResolvePseudoTypeClassType(IdentifierTypeNode $typeNode, NameScope $nameScope): ?Type
+	{
+		if (isset($nameScope->getUses()[strtolower($typeNode->name)])) {
```

Comment:
Instead of this leaking of NameScope internals, I'd add `NameScope::hasUseAlias()`.

Reply:
@ondrejmirtes good idea. This is done.